src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -25,48 +26,77 @@ internal TagHelperAttributeDescriptor([NotNull] string name, [NotNull] PropertyI
         /// <summary>
         /// Instantiates a new instance of the <see cref="TagHelperAttributeDescriptor"/> class.
         /// </summary>
-        /// <param name="name">The HTML attribute name.</param>
+        /// <param name="name">
+        /// The HTML attribute name or, if <paramref name="isIndexer"/> is <c>true</c>, the prefix for matching
+        /// attribute names.
+        /// </param>
         /// <param name="propertyName">The name of the CLR property that corresponds to the HTML attribute.</param>
         /// <param name="typeName">
-        /// The full name of the named (see <paramref name="propertyName"/>) property's <see cref="System.Type"/>.
+        /// The full name of the named (see <paramref name="propertyName"/>) property's <see cref="Type"/> or,
+        /// if <paramref name="isIndexer"/> is <c>true</c>, the full name of the indexer's value <see cref="Type"/>.
+        /// </param>
+        /// <param name="isIndexer">
+        /// If <c>true</c> this <see cref="TagHelperAttributeDescriptor"/> is used for dictionary indexer assignments.
+        /// Otherwise this <see cref="TagHelperAttributeDescriptor"/> is used for property assignment.
         /// </param>
+        /// <remarks>
+        /// HTML attribute names are matched case-insensitively, regardless of <paramref name="isIndexer"/>.
+        /// </remarks>
         public TagHelperAttributeDescriptor(
```

Comment:
If it makes it easier, i'd be ok with another internal ctor where it defaults `isIndexer` to `false`. If it's easier for you as is, leave it be :smile: